packages/cli/src/parseArgs.js

```diff
@@ -1,21 +1,19 @@
+const fs = require('fs')
+
 const { getDesignEntryPoint } = require('@jscad/core/code-loading/requireDesignUtilsFs')
 const { conversionFormats } = require('@jscad/core/io/formats')
-const fs = require('fs')
+
 const version = require('../package.json').version
 const env = require('./env')
 
 const parseArgs = args => {
   // hint: https://github.com/substack/node-optimist
   //       https://github.com/visionmedia/commander.js
-  //
-  // process.argv.forEach(function (val, index, array) {
-  //  console.log(index + ': ' + val)
-  // })
   if (args.length < 1) {
     console.log('USAGE:\n\nopenjscad [-v] <file> [-of <format>] [-o <output>]')
-    console.log('\t<file>  :\tinput (Supported types: folder, .jscad, .js, .scad, .stl, .amf, .obj, .gcode, .svg, .json, .dxf)')
-    console.log('\t<output>:\toutput (Supported types: folder, .jscad, .stl, .amf, .dxf, .svg, .json)')
-    console.log("\t<format>:\t'jscad', 'stla' (STL ASCII, default), 'stlb' (STL Binary), 'amf', 'dxf', 'svg', 'json'")
+    console.log('\t<file>  :\tinput (Supported types: folder, .jscad, .js, .amf, .dxf, .obj, .stl, .svg)')
```

Comment:
since we have a flat list of formats would it not make sense to auto generate these console outputs from the list ?

Reply:
i took a shot are generating lists of input extensions, output extensions, and output formats from the formats. see core/io/formats.js

the list of formats now controls what is deserializable as input, as well as what can be serialized as output.

the cli now uses those lists to log those lists, as well as build regular expressions.